src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -39,8 +45,42 @@
      * Андрей - 600
      * София - 900</p>
      */
+    private final SimpleDateFormat format = new SimpleDateFormat();
+
+    public SalaryInfo() {
+        format.applyPattern("dd.MM.yyyy");
+    }
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
             throws Exception {
-        return null;
+        Date startDate = format.parse(dateFrom);
+        Date endDate = format.parse(dateTo);
+        if (startDate.after(endDate)) {
+            throw new IllegalDateParametersException();
+        }
+        Map<String, Integer> totalSalary = new HashMap<>();
```

Comment:
Map is some kind of two-dimension array, but more complicated, so it's also not a good idea to use